mllib-dal/src/main/native/oneapi/dal/HomogenTableImpl.cpp

```diff
@@ -127,8 +127,8 @@ JNIEXPORT jlong JNICALL Java_com_intel_oneapi_dal_table_HomogenTableImpl_iInit(
     JNIEnv *env, jobject, jlong cRowCount, jlong cColCount, jintArray cData,
     jint cLayout, jint computeDeviceOrdinal) {
     printf("HomogenTable int init \n");
-    jboolean isCopy = true;
-    jint *fData = env->GetIntArrayElements(cData, &isCopy);
+    jboolean isCopy = JNI_FALSE;
+    jint *fData = static_cast<jint *>(env->GetPrimitiveArrayCritical(cData, &isCopy));
```

Comment:
Need to check if fData is NULL, https://docs.oracle.com/javase/8/docs/technotes/guides/jni/spec/functions.html#Get_PrimitiveType_ArrayElements_routines